### PR TITLE
pf ref commands should reject directory names passed for symlink creation

### PR DESCRIPTION
**Pull request by trstickland**
Usually the `-l`/`--symlink` option takes a directory name (e.g. when used with `pf data`), but with `pf ref` commands that causes an error.   If the users does this, it should cause an error with a suitable message.

(`pf ref` can take this option without a value, or with name that should be used for the new symlink.)

### CODE REVIEW

File: lib/Bio/Path/Find/App/PathFind/Ref.pm [hunk 1]
```diff
@@ -399,6 +399,25 @@ has '_paths' => (
 sub run {
   my $self = shift;
 
+  if ( $self->_symlink_flag and           # flag is set; we're making symlinks.
+       $self->_symlink_dest) and          # destination is specified.
+       -e $self->_symlink_dest ) {        # but destintation path exists
+    # this is an error, but tailor the message according to whether it is a directory
+    # (this matters because with other pf commands, the -l/--symlink option
+    # *should* provide a directory name; so it is a likely cause of confusuon for users)
```
Comment (Lfulcrum) [inline lib/Bio/Path/Find/App/PathFind/Ref.pm:407]:
```suggestion
    # *should* provide a directory name; so it is a likely cause of confusion for users)
```

**Review by Lfulcrum — CHANGES_REQUESTED** [review]

**Comment by trstickland** [comment]
Docker build fixed

**Comment by trstickland** [comment]
Docker build fixed

Requires dzil install uses `--notest`.  This requires tests have been run on devel machine, to be safe.

File: lib/Bio/Path/Find/App/PathFind/Ref.pm (hunk 1 above)
Comment (trstickland) [reply lib/Bio/Path/Find/App/PathFind/Ref.pm]: Fixed

**Comment by trstickland** [comment]
@Lfulcrum when this is merged and tagged we can check the automated docker build, to see if it is still being triggered

**Review by Lfulcrum — APPROVED** [review]
LGTM

**Pull request merged**